src/optable.ts

```diff
@@ -316,22 +316,30 @@ optable[opcodes.LOAD_CONST] = function(f: Py_FrameObject) {
     f.push(f.codeObj.consts[i]);
 }
 
-optable[opcodes.LOAD_NAME] = function(f: Py_FrameObject) {
+optable[opcodes.LOAD_NAME] = function(f: Py_FrameObject, t: Thread) {
     var i = f.readArg();
     var name = f.codeObj.names[i];
     var val = f.locals.get(name) || f.globals.get(name) || (<any> builtins)[`$${name.toString()}`];
+    // throw NameError
     if (val === undefined) {
-        throw new Error('undefined name: ' + name);
+        var message = `NameError: global name '${name}' is not defined\n`;
+        raise_exception_here(f, t, message, "NameError");
+        // get NameError type object
+        val = builtins['$NameError'];
```

Comment:
I haven't looked into Python exceptions too much; is it supposed to push the `NameError` type object onto the stack?